ui/src/members/components/AddMembersOverlay.tsx

```diff
@@ -35,36 +47,46 @@ export default class AddMembersOverlay extends PureComponent<Props, State> {
   }
 
   public render() {
-    const {onCloseModal} = this.props
+    const {status} = this.props
     const {selectedUserIDs, searchTerm, selectedMembers} = this.state
 
     return (
-      <Overlay.Container maxWidth={500}>
-        <Overlay.Heading title="Add Member" onDismiss={onCloseModal} />
-        <Overlay.Body>
-          <FilterList<User>
-            list={this.filteredList}
-            searchTerm={searchTerm}
-            searchKeys={['name']}
-          >
-            {ts => (
-              <AddMembersForm
-                onCloseModal={onCloseModal}
-                users={ts}
-                onSelect={this.handleSelectUserID}
-                selectedUserIDs={selectedUserIDs}
-                onSave={this.handleSave}
+      <Overlay visible={true}>
+        <Overlay.Container maxWidth={500}>
+          <Overlay.Heading title="Add Member" onDismiss={this.handleDismiss} />
+          <Overlay.Body>
+            <SpinnerContainer
+              loading={status}
+              spinnerComponent={<TechnoSpinner />}
+            >
+              <FilterList<User>
+                list={this.filteredList}
                 searchTerm={searchTerm}
-                onFilterChange={this.handleFilterChange}
-                onFilterBlur={this.handleFilterBlur}
-                selectedMembers={selectedMembers}
-              />
-            )}
-          </FilterList>
-        </Overlay.Body>
-      </Overlay.Container>
+                searchKeys={['name']}
+              >
+                {ts => (
+                  <AddMembersForm
+                    onCloseModal={this.handleDismiss}
+                    users={ts}
+                    onSelect={this.handleSelectUserID}
+                    selectedUserIDs={selectedUserIDs}
+                    onSave={this.handleSave}
+                    searchTerm={searchTerm}
+                    onFilterChange={this.handleFilterChange}
+                    onFilterBlur={this.handleFilterBlur}
+                    selectedMembers={selectedMembers}
+                  />
+                )}
+              </FilterList>
+            </SpinnerContainer>
+          </Overlay.Body>
+        </Overlay.Container>
+      </Overlay>
     )
   }
+  public componentDidMount() {
+    this.props.getUsers()
```

Comment:
Hrmm.. Could we use the GetResources component here instead since it handles the spinner container and everything. That way we have one component that handles fetching data?

Reply:
Hrmmm actually, I think you are already wrapping this component in a GetResources ? in `ui/src/members/containers/MembersIndex.tsx` below